shell/platform/android/io/flutter/embedding/engine/loader/FlutterLoader.java

```diff
@@ -158,8 +160,18 @@ public void startInitialization(@NonNull Context applicationContext, @NonNull Se
 
     initStartTimestampMillis = SystemClock.uptimeMillis();
     flutterApplicationInfo = ApplicationInfoLoader.load(appContext);
-    VsyncWaiter.getInstance((WindowManager) appContext.getSystemService(Context.WINDOW_SERVICE))
-        .init();
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
did it mean `TIRAMISU` ? https://source.corp.google.com/android/frameworks/base/core/java/android/os/Build.java;l=1137;bpv=0;bpt=1

Reply:
This check is specifically making sure the new code is used only for API 23+ which is when the display API was added. I guess it could be gated to only apply to T and above, I have no strong opinions either way. End goal is just to obtain the FPS float.

Reply:
ah ok, that's fine too then

Reply:
Gating at 23 is more likely to have better test coverage as most of our testing is with API 30 and below (as of now), making this safer in the long run. I'll leave it as is.

Reply:
The display API was actually added in 17. I'd like to change this, is there any good reason not to?

Reply:
Or at least, all of the display code we're using :)

Reply:
I'm changing this in #29000 - please add comments if it needs to be at 23 for some reason I'm missing.

Reply:
That should hav ebeen #29800....

Reply:
I see now - getSystemService was using the API 23 call instead of the one compatible with API 17. Fixing that upstream in my PR.